pkg/controller/argocd/secret.go

```diff
@@ -379,6 +386,50 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	secret := argoutil.NewSecretWithSuffix(cr.ObjectMeta, "default-cluster-config")
+	secret.Labels[common.ArgoCDSecretTypeLabel] = "cluster"
+	dataBytes, _ := json.Marshal(map[string]interface{}{
+		"tlsClientConfig": map[string]interface{}{
+			"insecure": false,
+		},
+	})
+
+	secret.Data = map[string][]byte{
+		"config":     dataBytes,
+		"name":       []byte("in-cluster"),
+		"server":     []byte(common.ArgoCDDefaultServer),
+		"namespaces": []byte(cr.Namespace),
+	}
+
+	applyReconcilerHook(cr, secret, "")
+
+	clusterSecrets := &corev1.SecretList{}
+	opts := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			common.ArgoCDSecretTypeLabel: "cluster",
+		}),
+		Namespace: cr.Namespace,
+	}
+
+	if err := r.client.List(context.TODO(), clusterSecrets, opts); err != nil {
+		return err
+	}
+
+	secrets := clusterSecrets.Items
+
+	for _, s := range secrets {
+		// check if cluster secret with name in-cluster exists
+		// do nothing if exists.
+		if string(s.Data["name"]) == "in-cluster" {
```

Comment:
It is better to find by server address than by name because the name can changed and does not have to be "in-cluster".

Reply:
No i think the point here is to not have multiple secrets for `in-cluster` configuration. The name has to be `in-cluster` AFAIK. This actually is not the name of the secret, it is infact the name of the cluster configuration, which i am assume will always be `in-cluster`. 
@jannfis can maybe provide a better insight on this!

Reply:
@wtam2018 has a good point there, the user may change the symbolic name of the default cluster from `in-cluster` to something else. The constant here is the server address, `https://kubernetes.default.svc`.

Reply:
Thanks @jannfis @wtam2018 !! fixed it !